docs: add documentation for Pyroscope SDK profiles through Alloy (#3658)
grafana/pyroscope: docs/sources/configure-client/grafana-alloy/_index.md

1. Ensure Alloy is set to pull mode and targeting the correct pprof endpoints. For step-by-step instructions, visit the [Go (Pull Mode)](https://grafana.com/docs/pyroscope/<PYROSCOPE_VERSION>/configure-client/grafana-alloy/go_pull) documentation.
1. The collector queries the pprof endpoints of your Golang application, collects the profiles, and forwards them to the Pyroscope server.

> ## Receiving profiles from Pyroscope SDKs

Reviewer: We've moved away from using -ing verbs in headings. We use imperative forms: Receive instead of Receiving. For more information, https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/
```suggestion
## Receive profiles from Pyroscope SDKs
```

> Alloy (not available in Grafana Agent) can receive profiles from applications instrumented with Pyroscope SDKs through the `pyroscope.receive_http` component. This approach provides several key advantages:
> - Improved performance by sending profiles to a local Alloy instance instead of over the internet to Grafana Cloud, reducing latency and application impact
> - Separation of infrastructure concerns from application code - developers don't need to handle authentication, tenant configuration, or infrastructure labels in their code
> - Centralized management of authentication and metadata enrichment (for example, Kubernetes labels, business labels)

Reviewer: We try to avoid using parenthetical statements. It felt like having the Agent statement up front took away from the focus on Alloy. I've added a statement below noting that the capability isn't available in the legacy Grafana Agent. What do you think?
```suggestion
Alloy can receive profiles from applications instrumented with Pyroscope SDKs through the `pyroscope.receive_http` component. This approach provides several key advantages:
- Improved performance by sending profiles to a local Alloy instance instead of over the internet to Grafana Cloud, reducing latency and application impact
- Separation of infrastructure concerns from application code - developers don't need to handle authentication, tenant configuration, or infrastructure labels in their code
- Centralized management of authentication and metadata enrichment (for example, Kubernetes labels, business labels)

This capability is not available in the legacy Grafana Agent.
```

### Set up profile receiving

> 1. Configure your application with a Pyroscope SDK pointing to Alloy's receive endpoint

Reviewer: We try not to use possessives with product names. 
```suggestion
1. Configure your application with a Pyroscope SDK pointing to receive endpoint for Alloy. 
```

> 2. For step-by-step instructions, visit the [Receiving SDK profiles](https://grafana.com/docs/pyroscope/<PYROSCOPE_VERSION>/configure-client/grafana-alloy/receive_profiles) documentation.

Reviewer: ```suggestion
2. For step-by-step instructions, refer to the [Receive profiles from Pyroscope SDKs](https://grafana.com/docs/pyroscope/<PYROSCOPE_VERSION>/configure-client/grafana-alloy/receive_profiles) documentation.
```

1. Ensure Alloy is set to pull mode and targeting the correct pprof endpoints. For step-by-step instructions, visit the [Go (Pull Mode)](https://grafana.com/docs/pyroscope/<PYROSCOPE_VERSION>/configure-client/grafana-alloy/go_pull) documentation.
1. The collector queries the pprof endpoints of your Golang application, collects the profiles, and forwards them to the Pyroscope server.

## Receive profiles from Pyroscope SDKs

Alloy can receive profiles from applications instrumented with Pyroscope SDKs through the `pyroscope.receive_http` component. This approach provides several key advantages:

> - Improved performance by sending profiles to a local Alloy instance instead of over the internet to Grafana Cloud, reducing latency and application impact
> - Separation of infrastructure concerns from application code - developers don't need to handle authentication, tenant configuration, or infrastructure labels in their code
> - Centralized management of authentication and metadata enrichment (for example, Kubernetes labels, business labels)

Reviewer: This suggestion makes the bullet structure parallel. 
```suggestion
- Improved performance by sending profiles to a local Alloy instance instead of over the internet to Grafana Cloud, which reduces latency and application impact
- Separated infrastructure concerns from application code so developers don't need to handle authentication, tenant configuration, or infrastructure labels in their code
- Centralized management of authentication and metadata enrichment (for example, Kubernetes labels, business labels)
```
